nodelet_topic_tools/include/nodelet_topic_tools/nodelet_lazy.h

```diff
@@ -144,9 +146,28 @@ class NodeletLazy: public nodelet::Nodelet
     if (lazy_)
     {
       boost::mutex::scoped_lock lock(connection_mutex_);
-      for (size_t i = 0; i < publishers_.size(); i++)
+      BOOST_FOREACH (const ros::Publisher& pub, publishers_)
+      {
+        if (pub.getNumSubscribers() > 0)
```

Comment:
This code is the same for both types of subscribers.
I can factorize it into a templated method to avoid code duplication. I didn't do it b/c I think this way the intention and changes are cleaner in the diff. I can do that in another PR.